gdnative-core/src/geom/basis.rs

```diff
@@ -19,8 +25,89 @@ impl Basis {
         unsafe { std::mem::transmute::<sys::godot_basis, Self>(c) }
     }
 
-    pub fn identity() -> Basis {
+    /// The basis that will flip something along the **X Axis** when used in a transformation.
+    pub const fn flip_x() -> Basis {
+        Basis::from_diagonal(Vector3::new(-1.0, 1.0, 1.0))
+    }
+
+    /// The basis that will flip something along the **Y axis** when used in a transformation.
+    pub const fn flip_y() -> Basis {
+        Basis::from_diagonal(Vector3::new(1.0, -1.0, 1.0))
+    }
+
+    /// The basis that will flip something along the **Z axis** when used in a transformation.
+    pub const fn flip_z() -> Basis {
+        Basis::from_diagonal(Vector3::new(1.0, 1.0, -1.0))
+    }
+
+    /// Creates a Basis from the given [`Vector3`](./type.Vector3.html)
+    pub const fn from_diagonal(p_diag: Vector3) -> Basis {
         Basis {
+            elements: [
+                Vector3::new(p_diag.x, 0.0, 0.0),
+                Vector3::new(0.0, p_diag.y, 0.0),
+                Vector3::new(0.0, 0.0, p_diag.z),
+            ],
+        }
+    }
+
+    /// Creates a Basis from 3 [`Vectors3`](./type.Vector3.html)s
+    pub fn from_elements(elements: [Vector3; 3]) -> Self {
+        Self { elements }
+    }
+
+    /// Creates a rotation matrix.
+    pub fn from_euler(euler: Vector3) -> Self {
+        let mut b = Basis::default();
+        b.set_euler_yxz(&euler);
+        b
+    }
+
+    /// Rotation matrix from axis and angle.
+    ///
+    /// See [https://en.wikipedia.org/wiki/Rotation_matrix#Rotation_matrix_from_axis_angle]()
```

Comment:
Are you sure that this link works? I don't think it does what one would expect it to do according to GFM specs: https://github.github.com/gfm/#example-495

Also, the doc-comments for this method may need a `Panics` section, seeing that it can panic if `axis` is not normalized:

```rust
/// ...
///
/// # Panics
///
/// If `axis` is not normalized.
```